apps/tangle-dapp/data/restake/useOperatorTVL.ts

```diff
@@ -0,0 +1,66 @@
+import { useObservable, useObservableState } from 'observable-hooks';
+import { of, switchMap } from 'rxjs';
+
+import type { AssetMap, OperatorMap } from '../../types/restake';
+import safeFormatUnits from '../../utils/safeFormatUnits';
+
+export function useOperatorTVL(operatorMap: OperatorMap, assetMap: AssetMap) {
+  const tvl$ = useObservable(
+    (input$) =>
+      input$.pipe(
+        switchMap(([operatorMap, assetMap]) => {
+          const { operatorTVL, vaultTVL } = Object.entries(operatorMap).reduce(
+            (acc, [operatorId, operatorData]) => {
+              const operatorTVL = operatorData.delegations.reduce(
+                (sum, delegation) => {
+                  const asset = assetMap[delegation.assetId];
+                  const assetPrice = asset?.priceInUsd ?? null;
+
+                  if (typeof assetPrice !== 'number') {
+                    return sum;
+                  }
+
+                  const result = safeFormatUnits(
+                    delegation.amount,
+                    asset.decimals,
+                  );
+
+                  if (!result.sucess) {
+                    return sum;
+                  }
+
+                  const formattedAmount = Number(result.value);
```

Comment:
Would `amount` be a better name? Formatted amount might imply that it's in string form. What do you think?